spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2443,6 +2446,9 @@ private void deliveryAttemptHeader(final ConsumerRecord<K, V> record) {
 								new TopicPartitionOffset(record.topic(), record.partition(), record.offset())));
 				record.headers().add(new RecordHeader(KafkaHeaders.DELIVERY_ATTEMPT, buff));
 			}
+			if (this.listenerinfo != null) {
+				record.headers().add(this.infoHeader);
```

Comment:
Why cannot we do conversion to `byte[]` over here?
I mean that is really just an `info`, so end-users will expect from us a plain `public void setListenerInfo(@Nullable String listenerInfo) {` from the configuration side.